src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -314,9 +328,14 @@ def run(self, prompts: Prompts, include_names: set[str] | None = None) -> list[S
         )
         sp_results = []
         if plan_confirmed:
-            sp_results = self._migrate_service_principals(sp_migration_infos)
+            try:
+                sp_results = self._migrate_service_principals(sp_migration_infos)
+            except (BadRequest, PermissionDenied, NotFound):
+                self._resource_permissions.delete_storage_credentials(execution_results)
```

Comment:
Do we need to delete the access connectors if the SP migration fails? I would say that we prefer to stick with the access connectors in that case

Reply:
yes that makes sense. removing the code for access connectors roll back in this step